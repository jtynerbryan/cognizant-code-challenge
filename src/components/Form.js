import React from 'react';
import Input from './Input'

class Form extends React.Component {

  state = {
    apiKey: 'ce658e43d986d3db6fdafac97e5a006b',
    inputLength: 0,
    selectedCities: [],
    cities: []
  }

  componentDidMount() {
    // fetch('http://localhost:3000/cities?name=Boston')
    // .then(res => res.json())
    // .then(res => console.log(res))


    // fetch('http://api.openweathermap.org/data/2.5/forecast?id=4317656&APPID=' + this.state.apiKey + '&units=imperial')
    // .then(res => res.json())
    // .then(res => console.log(res))
  }

  handleNumSelect = (e) => {
    console.log(e.target.value);
    e.preventDefault()
    if (parseInt(e.target.value, 10) < 4) {
      alert('Please enter a valid number no less than 4')
    }
    this.setState({
      inputLength: e.target.value
    })
  }

  render() {
    if (this.state.inputLength < 4) {
      return (
        <div style={{margin: 'auto', width: '50%', textAlign: 'center'}}>
          <h1>Select how many cities you would like to find weather for (minimum of 4)</h1>
          <form >
            <input onChange={this.handleNumSelect} type='text' style={{borderColor: 'black', borderRadius: '5px', marginRight: '5px', fontSize: '20px', width: '30px'}}/>
          </form>
        </div>
      )
    } else {
      let inputs = []
      for (let i = 0; i < this.state.inputLength; i++) {
        inputs.push(<div key={i}><Input/></div>)
      }
      return (
        <div style={{margin: 'auto', width: '50%', textAlign: 'center'}}>
          <h1>Enter cities to find weather for:</h1>
          <form>
            {inputs}
          </form>
        </div>
      )
    }
  }
}

export default Form
