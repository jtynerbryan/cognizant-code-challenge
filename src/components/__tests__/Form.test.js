import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../Form';

describe('<Form/>', () => {
  it('renders without crashing', () => {
    shallow(<Form />);
  })

  it('renders user instructions', () => {
    const wrapper = shallow(<Form />);
    const instructions = <h2>Please enter at least 4 cities to find weather data for</h2>
    expect(wrapper.contains(instructions)).toEqual(true);
  });

  it('renders without changing unexpectedly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
