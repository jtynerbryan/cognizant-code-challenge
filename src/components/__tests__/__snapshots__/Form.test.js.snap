// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders without changing unexpectedly 1`] = `
<div
  style={
    Object {
      "margin": "auto",
      "marginTop": "10px",
      "textAlign": "center",
      "width": "50%",
    }
  }
>
  <h2>
    Please enter at least 4 cities to find weather data for
  </h2>
  <h3>
    Seperate each city by comma (ex. 'Boston, New York, London, Berlin')
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      style={
        Object {
          "borderColor": "black",
          "borderRadius": "5px",
          "fontSize": "15px",
          "height": "20px",
          "marginBottom": "10px",
          "width": "300px",
        }
      }
      type="text"
      value=""
    />
    <input
      style={
        Object {
          "borderColor": "black",
          "borderRadius": "5px",
          "height": "25px",
        }
      }
      type="submit"
    />
  </form>
</div>
`;
